src/code/z_snap.c

```diff
@@ -119,42 +146,48 @@ s32 func_8013A530(PlayState* play, Actor* actor, s32 flag, Vec3f* pos, Vec3s* ro
     s32 ret = 0;
     s32 bgId;
 
+    // Check distance
     distance = OLib_Vec3fDist(pos, &camera->eye);
     if ((distance < distanceMin) || (distanceMax < distance)) {
-        func_8013A41C(0x3F);
-        ret = 0x3F;
+        Snap_SetFlag(PICTOGRAPH_BAD_DISTANCE);
+        ret = PICTOGRAPH_BAD_DISTANCE;
     }
 
-    x = func_8013A504(Camera_GetCamDirPitch(camera) + rot->x);
-    y = func_8013A504(Camera_GetCamDirYaw(camera) - BINANG_SUB(rot->y, 0x7FFF));
-    if ((0 < angleError) && ((angleError < x) || (angleError < y))) {
-        func_8013A41C(0x3E);
-        ret |= 0x3E;
+    // Check rot is facing camera?
+    x = Snap_AbsS(Camera_GetCamDirPitch(camera) + rot->x);
+    y = Snap_AbsS(Camera_GetCamDirYaw(camera) - BINANG_SUB(rot->y, 0x7FFF));
+    if ((0 < angleRange) && ((angleRange < x) || (angleRange < y))) {
+        Snap_SetFlag(PICTOGRAPH_BAD_ANGLE);
+        ret |= PICTOGRAPH_BAD_ANGLE;
     }
 
-    Actor_GetProjectedPos(play, pos, &screenSpace, &distance);
-    x = (s16)(screenSpace.x * distance * 160.0f + 160.0f) - 85;
-    y = (s16)(screenSpace.y * distance * -120.0f + 120.0f) - 67;
-    if ((x < 0) || (150 < x) || (y < 0) || (105 < y)) {
-        func_8013A41C(0x3D);
-        ret |= 0x3D;
+    // Check in viewfinder
+    Actor_GetProjectedPos(play, pos, &projectedPos, &distance);
+    x = (s16)(projectedPos.x * distance * 160.0f + 160.0f) - 85;
+    y = (s16)(projectedPos.y * distance * -120.0f + 120.0f) - 67;
+    if ((x < 0) || (x > 150) || (y < 0) || (y > 105)) {
+        Snap_SetFlag(PICTOGRAPH_NOT_IN_VIEW);
+        ret |= PICTOGRAPH_NOT_IN_VIEW;
     }
 
-    if (BgCheck_ProjectileLineTest(&play->colCtx, pos, &camera->eye, &screenSpace, &poly, true, true, true, true,
+    // Check not obscured by bg collision
+    if (BgCheck_ProjectileLineTest(&play->colCtx, pos, &camera->eye, &projectedPos, &poly, true, true, true, true,
                                    &bgId)) {
-        func_8013A41C(0x3C);
-        ret |= 0x3C;
+        Snap_SetFlag(PICTOGRAPH_BEHIND_BG);
+        ret |= PICTOGRAPH_BEHIND_BG;
     }
 
+    // Check not obscured by actor collision
     actors[0] = actor;
     actors[1] = &GET_PLAYER(play)->actor;
     if (CollisionCheck_LineOCCheck(play, &play->colChkCtx, pos, &camera->eye, actors, 2)) {
-        func_8013A41C(0x3B);
-        ret |= 0x3B;
+        Snap_SetFlag(PICTOGRAPH_BEHIND_COLLISION);
+        ret |= PICTOGRAPH_BEHIND_COLLISION;
     }
 
+    // If all of the above checks pass, set the flag
     if (ret == 0) {
```

Comment:
```suggestion
    // If all of the above checks pass, set the flag
    if (errorFlags == 0) {
```
Perhaps?

Reply:
I think ret's fine, this is pretty much how you usually use ret. It's also not returning flags anyway.

Reply:
I've never been a fan of `ret` to begin with, it's only a slight step up from the generic `temp` imo. It's appropriate in libultra where the goal is to retain names that the original devs used, fair enough. But if there's something meaningful to the ret, it should be named (might just be me though).

Is it not returning flags? It's returning values from the `PictographFlag` enum, specifically the error flags with the `0x20` involved

Reply:
It returns an or'd-together mess that you can't interpret apart from "it failed"

Reply:
Actually yeah you're right. It's confusing because the enum isn't the flags themselves, it's the index for the shift of the flags i.e. `1 << enumVal`, but it's `|= enumVal` in this return. Fair enough. I wonder if `PictographFlag` is fine since it's not really a flag, but the shift value for a flag. It's probably fine though

Reply:
I think it's consistent with what we do elsewhere for things like the flag functions in z_actor, where the `flag` input is the shift/index. But this is probably precisely what confused them in the first place, and it does make it very inconvenient to find out how a pictograph failed, since the actual flags in SaveContext are volatile and change value every time an actor runs this function. I'll add to the note, and we might want to consider how to differentiate these in future.